v2/internal/carv1/car.go

```diff
@@ -209,3 +210,8 @@ func loadCarSlow(s Store, cr *CarReader) (*CarHeader, error) {
 		}
 	}
 }
+
+func (h CarHeader) Equals(other CarHeader) bool {
+	// TODO should headers with same roots in different order be equal?
```

Comment:
i'd probably lean towards 'yes' - or at the least i think we have a canonical order they should get written into and in practice it won't matter since there are very very few cases where there isn't exactly one root